### PR TITLE
[ruby-mentoring] - Add Test coverage

### PR DESCRIPTION
**Pull request by roanhawkeye**
- Add definitions for User and Movie for Faker,
- Add simple unit test for model
- Add request testing for index and create methods.
- Add validations to model movie just to simplify testing for educational purpose - Ideally, I should add a Movie Form to validates and abstract validations from the model.

### CODE REVIEW

File: spec/request/movie_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      post '/movies', :params => { :movie => { 
+        imdb_id: 'some_imdb_id', 
+        title: 'Hobbit' , 
+        rating: 4.5, 
+        rank: 10, 
+        year: 2010 } 
+        } 
+    }
+    
+    it { expect(response).to have_http_status(302)}
```
Comment (evgeny-danilov) [inline spec/request/movie_request_spec.rb:25]: Our requests specs are quite heavy. As long as we tend to follow the Pyramid of tests for our Ruby projects, it's better to join this expectation with the next one. In this way we'll send only one request in specs instead of two.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe User, :type => :model do
+  let(:user) {FactoryBot.create :user}
+
+  it "when valid" do
+    expect(user).to be_valid  
+  end
```
Comment (evgeny-danilov) [inline spec/models/user_spec.rb:8]: This is kind of a unit test, which is quite lightweight. So, we can test all possible edge cases of the class. 
In this particular case it means we can add at least three expectations. So, the ultimate result of the command `rspec --format=documentation` will look like this
```
when validate the model
  when attributes are present
    does not raise an error
  when user is empty
    raises a validation error
  when title is empty
    raises an validation error
```

As you can see, our tests can work as an up-to-date documentation.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe User, :type => :model do
+  let(:user) {FactoryBot.create :user}
+
+  it "when valid" do
```
Comment (evgeny-danilov) [inline spec/models/user_spec.rb:6]: The general rule is that with `context` we use keywords `when`, and with `it` a real action. For example:
```
context 'when there is some context`
  it 'creates a record in the DB`
     expect(DB).to create(some_record)
  end
end

File: spec/request/movie_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    it 'increase movies count' do
+      expect {
+        post '/movies', :params => { :movie => { 
+          imdb_id: 'some_imdb_id', 
+          title: 'Hobbit' , 
+          rating: 4.5, 
+          rank: 10, 
+          year: 2010 } 
+        } 
+      }.to change{ Movie.count }.by(1)
```
Comment (evgeny-danilov) [inline spec/request/movie_request_spec.rb:36]: It seems after this expectation there will be a few records in the DB (as you do two post-requests - one from `before` block, and one from `it` block). So, the expectation is correct, but the performance is not optimal.

File: spec/request/movie_request_spec.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    it { expect(response).to have_http_status(:ok) }
+  end
+
+end
+
+describe '#create', :type => :request do
+  context 'when valid' do
+    
+    before {
+      post '/movies', :params => { :movie => {
```
Comment (evgeny-danilov) [inline spec/request/movie_request_spec.rb:16]: [nitpick] Since Ruby 2.x we have nicer syntax with less symbols, as we're are lazy to type a lot :)
```suggestion
      post '/movies', params: { movie: { 
```
BTW, have you check the last changes with Rubocop linter?

**Comment by evgeny-danilov** [comment]
In general, everything looks good 👍
